lib/rubocop/cop/style/multiline_list.rb

```diff
@@ -0,0 +1,158 @@
+      # multi-line list of method parameters, method arguments, array
+      # elements, or hash elements.
+      #
+      # The cop start by looking for any of the following:
+      #
+      # * The first parameter in a method definition.
+      # * The first argument in a method call.
+      # * The first element in an array.
+      # * The first element in a hash.
+      #
+      # when the list of parameters, arguments or elements spans multiple lines
+      # It then ensures that the first such element does not also occur on the
+      # object's first line.
+      #
+      # It supports three configuration parameters:
+      #
+      # * EnableMethodParams - boolean
+      # * EnableMethodArgs - boolean
+      # * EnableHashElements - boolean
+      # * EnableArrayElements - boolean
+      #
+      # @example
+      #
+      #     # bad with EnableMethodParams
+      #     def method(foo, bar,
+      #         baz)
+      #       do_something
+      #     end
+      #
+      #     # good with EnableMethodParams
+      #     def method(
+      #         foo, bar,
+      #         baz)
+      #       do_something
+      #     end
+      #
+      #     # ignored with EnableMethodParams
+      #     def method foo,
+      #         bar
+      #       do_something
+      #     end
+      #
+      #     # bad with EnableMethodArgs
+      #     method(foo, bar,
+      #       baz)
+      #
+      #     # good with EnableMeethodArgs
+      #     method(
+      #       foo, bar,
+      #       baz)
+      #
+      #     # ignored with EnableMethodArgs
+      #     method foo, bar,
+      #       baz
+      #
+      #     # bad with EnableHashElements
+      #     { a: 1,
+      #       b: 2}
+      #
+      #     # good with EnableHashElements
+      #     {
+      #       a: 1,
+      #       b: 2 }
+      #
+      #     # bad with EnableArrayElements
+      #     [ :a,
+      #       :b]
+      #
+      #     # good with EnableArrayElements
+      #     [
+      #       :a,
+      #       :b]
+      #
+      class MultilineList < Cop
+        include OnMethodDef
+
+        MSG = 'Add a line break before the first element of a multi-line list.'
+
+        def on_method_def(node, _method_name, args, _body)
+          return unless cop_config['EnableMethodParams']
+
+          check_method(node, args.to_a)
+        end
+
+        def on_send(node)
+          return unless cop_config['EnableMethodArgs']
+
+          _receiver, _name, *args = *node
+
+          check_method(node, args)
+        end
+
+        def on_array(node)
+          return unless cop_config['EnableArrayElements']
+
+          return if !node.loc.begin && !assignment_on_same_line?(node)
+
+          check_children(node, node.children)
+        end
+
+        def on_hash(node)
+          return unless cop_config['EnableHashElements']
+
+          if node.loc.begin
+            check_children(node, node.children)
+          elsif !method_uses_parens?(node.parent, node)
```

Comment:
Can the `elsif` and the `else` be flipped here? Assuming this doesn't break anything, I think it would be more readable as

```ruby
elsif method_uses_parens?(node.parent, node)
  check_children(node, node.children, node.parent)
else
  return
end
```